### DISCUSSION TITLE
Is it possible to reference the pattern of a string

### DISCUSSION THREAD

**Discussion by robp94** [OAI/OpenAPI-Specification#3330] [Q&A] [answered]
If you need a pattern multiple times, I would like to reduce duplication. Is it possible to reference a pattern for a string?

**Comment by handrews** [comment, marked as answer]
You can just have a schema that only has `{"pattern": "the pattern"}` and use `"$ref"` to include it.

**Reply by robp94** [reply]
This does indeed work, even if intellij complains. What is a bit strange, the kotlin generator which I am using needs now `\\\\` sometimes instead of `\\`.

**Reply by robp94** [reply]
Hm no unfortunately this does not work, or at least my java and kotlin generators now use object instead of string.

**Reply by handrews** [reply]
That's probably something to take up with the tooling vendor.  While there's no formal spec for code generation from schemas, it's extremely common to stitch together disparate elements using `"allOf"` without each schema object being a distinct code object.  And for exactly the reasons you have - it's an implementation detail to avoid duplication within your schemas.

**Comment by handrews** [comment]
Closing as this is resolved from the specification perspective (or as resolved as it can be- OAS 3.x cannot change JSON Schema's `$ref`'s behavior)
